signer/src/error.rs

```diff
@@ -3,11 +3,31 @@ use std::borrow::Cow;
 
 use blockstack_lib::types::chainstate::StacksBlockId;
 
-use crate::{codec, ecdsa, network};
+use crate::{codec, context::SignerSignal, ecdsa, network};
 
 /// Top-level signer error
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
+    /// Error broadcasting a message to the application channel
+    #[error("Error broadcasting message to application channel: {0}")]
+    BroadcastSendError(#[from] tokio::sync::broadcast::error::SendError<SignerSignal>),
+
+    /// Error receiving a message from the application channel
+    #[error("Error receiving message from application channel: {0}")]
+    BroadcastRecvError(#[from] tokio::sync::broadcast::error::RecvError),
+
+    /// Error sending a message to the application termination channel
+    #[error("Error sending message to application termination channel: {0}")]
+    TerminationSendError(#[from] tokio::sync::watch::error::SendError<bool>),
+
+    /// Error sending a message to the application termination channel
+    #[error("Error receiving message from application termination channel: {0}")]
+    TerminationRecvError(#[from] tokio::sync::watch::error::RecvError),
```

Comment:
The same goes for these types. I think we can explicitly handle all failure cases in the code and should not re-propagate them.

Reply:
Removed these in [376cace](https://github.com/stacks-network/sbtc/pull/473/commits/376cacef68fb40239e51df5049feabaeed3189f6).

I changed these to simply be swallowed because if there's a send error, then we're already going down, and a receive error is impossible (the one place it's being used).

Reply:
Great. Note that we use these errors as a shutdown signal in many other components right now, which I think is pretty nice because it means that dropping a channel to one of these actors shuts it down without requiring an explicit shut-down mechanism. This could be an alternative to using a termination handle altogether.

Reply:
But I'm okay with having an explicit termination handle as well.